### PR TITLE
Support for transpose/dilated_lhs convolution for VMLA

### PR DESCRIPTION
**Pull request by rsuderman**
Includes conditionally disabling padding extraction.

### CODE REVIEW

File: iree/compiler/Dialect/Flow/Transforms/HLOToHLOPreprocessing.cpp [hunk 1]
```diff
@@ -114,6 +114,16 @@ class ExtractConvOpPaddingAttributes : public OpRewritePattern<mhlo::ConvOp> {
     if (!hasPadding(op)) return failure();
     auto inputType = op.lhs().getType().cast<ShapedType>();
     int rank = inputType.getRank();
+
+    // We can't extract padding if the left hand side has dilation.
+    if (op.lhs_dilation().hasValue()) {
```
Comment (asaadaldien) [inline iree/compiler/Dialect/Flow/Transforms/HLOToHLOPreprocessing.cpp:119]: This means we can't codegen convtranspoe: Lets add a TODO here, we have to figure out either:
1) Have codegen lowering for conv + padding
2) Extract correct padding argument in the transpose_conv case

**Review by asaadaldien — APPROVED** [review]
Thanks for fixing the dilation indexing @rsuderman. Can you please update the documentation of Conv2D in VMLA that it does also transpose_conv.

File: iree/compiler/Dialect/Flow/Transforms/HLOToHLOPreprocessing.cpp (hunk 1 above)
Comment (rsuderman) [reply iree/compiler/Dialect/Flow/Transforms/HLOToHLOPreprocessing.cpp:120]: Updated.

Comment (rsuderman) [reply iree/compiler/Dialect/Flow/Transforms/HLOToHLOPreprocessing.cpp:120]: (I should be able to handle this better in the follow up)

**Pull request merged**